### PR TITLE
feat(IFCImporter): add vtkIFCImporter

### LINKED ISSUE #3156: [Feature] Support for IFC format
### Motivation

Support for the Industry Foundation Classes (IFC) format.

### Detailed Description

Add a new vtkIFCImporter

```js
vtkResourceLoader
    .loadScript('https://cdn.jsdelivr.net/npm/web-ifc@0.0.55/web-ifc-api-iife.js')
    .then(() => {
        vtkIFCImporter.setIFCAPI(window.WebIFC);

        importer
            .setUrl('http://example.com/model.ifc');

        importer.onReady(() => {
            importer.getActors().forEach((actor) => renderer.addActor(actor));
            resetCamera();
            render();
        });
    });
```

### PR DESCRIPTION
**Pull request by daker**
This PR adds support for IFC format, the importer allows to merge geometries to reduce the number of actors in the scene.

The importer is using the [web-ifc](https://github.com/ThatOpen/engine_web-ifc) wasm module to parse & extract meshes

### Context
fixes #3156

### Changes
- [x] Documentation and TypeScript definitions were updated to match those changes

### PR and Code Checklist
- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing
- [x] This change adds or fixes unit tests 
- [x] Tested environment:
  - **vtk.js**: 
  - **OS**: 
  - **Browser**:

### CODE REVIEW

File: Sources/IO/Geometry/IFCImporter/index.js [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+import vtkMapper from 'vtk.js/Sources/Rendering/Core/Mapper';
+import vtkDataArray from 'vtk.js/Sources/Common/Core/DataArray';
+import vtkPolyData from 'vtk.js/Sources/Common/DataModel/PolyData';
+import vtkCellArray from 'vtk.js/Sources/Common/Core/CellArray';
+import vtkAppendPolyData from 'vtk.js/Sources/Filters/General/AppendPolyData';
+import vtkMatrixBuilder from 'vtk.js/Sources/Common/Core/MatrixBuilder';
+
+let WebIFC;
+
+function setIFCAPI(api) {
```
Comment (finetjul) [inline Sources/IO/Geometry/IFCImporter/index.js:20]: Should be documented, it's not clear what this is for

**Comment by finetjul** [comment]
> @finetjul i still need to fix the normals transformation when the geometries are merged, if you have any hints.

I think Matrix builder has a function to transform vectors(normals)

File: Sources/IO/Geometry/IFCImporter/example/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    render();
+  });
+}
+
+// ----------------------------------------------------------------------------
+// Import usage example
+// ----------------------------------------------------------------------------
+
+vtkResourceLoader
+  .loadScript('https://cdn.jsdelivr.net/npm/web-ifc@0.0.55/web-ifc-api-iife.js')
```
[the remark this answers is not in the document]
Comment (daker) [reply Sources/IO/Geometry/IFCImporter/example/index.js:39]: i don't have an account, can you upload that and share the link ?

**Review by finetjul — CHANGES_REQUESTED** [review]

File: Sources/IO/Geometry/IFCImporter/index.js [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+import vtkCellArray from 'vtk.js/Sources/Common/Core/CellArray';
+import vtkAppendPolyData from 'vtk.js/Sources/Filters/General/AppendPolyData';
+import vtkMatrixBuilder from 'vtk.js/Sources/Common/Core/MatrixBuilder';
+
+let WebIFC;
+
+/**
+ * Set WebIFC api to be used by vtkIFCImporter
+ * @param {object} ifcApi
+ */
```
[the remark this answers is not in the document]
Comment (daker) [reply Sources/IO/Geometry/IFCImporter/index.js:23]: it's already there

https://github.com/Kitware/vtk-js/pull/3168/files#diff-12358842b72292fad557d196b017e2c81f49c03965de4099cf9ab7f409df9a4dR104

File: Data/ifc/house.ifc
[the remark this answers is not in the document]
Comment (daker) [reply Data/ifc/house.ifc:1]: I did a demo file from the Web IFC repo directly

File: Sources/IO/Geometry/IFCImporter/index.js [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+        mapper.setInputData(pd);
+
+        actor = vtkActor.newInstance();
+        actor.setMapper(mapper);
+
+        const { x, y, z, w } = mesh.color;
+        actor.getProperty().setColor(x, y, z);
+        actor.getProperty().setOpacity(w);
+
+        actor.setUserMatrix(mesh.userMatrix);
```
[the remark this answers is not in the document]
Comment (daker) [reply Sources/IO/Geometry/IFCImporter/index.js:292]: i did apply the matrix in `createColoredPolyDataFromIFCMesh`

**Review by finetjul — CHANGES_REQUESTED** [review]
Can you please add a screenshot in the example gallery.

**Comment by daker** [comment]
@finetjul it's ready

**Review by finetjul — APPROVED** [review]
LGTM

**Comment by daker** [comment]
@floryst this is ready too :) i have another PR waiting for your review #3170

**Review by floryst — APPROVED** [review]

**Pull request merged**